sys/net/gnrc/routing/rpl/gnrc_rpl_dodag.c

```diff
@@ -309,9 +309,7 @@ static gnrc_rpl_parent_t *_gnrc_rpl_find_preferred_parent(gnrc_rpl_dodag_t *doda
         return NULL;
     }
 
-    LL_FOREACH(dodag->parents, elt) {
-        new_best = dodag->instance->of->which_parent(new_best, elt);
-    }
+    LL_SORT(dodag->parents, dodag->instance->of->which_parent);
```

Comment:
good point, then it should be done when ever the rank of a parent changes - what could be quite often and would introduce lots of overhead, which we don't want on an embedded node.

OK, I see this is not as easy as I first thought 😉